pkg/docker/docker.go

```diff
@@ -132,6 +132,55 @@ type RunContainerOptions struct {
 	CapDrop          []string
 }
 
+// asDockerConfig converts a RunContainerOptions into a Config understood by the
+// docker client library.
+func (rco RunContainerOptions) asDockerConfig() docker.Config {
+	return docker.Config{
+		Image: getImageName(rco.Image),
+		User:  rco.User,
+		Env:   rco.Env,
+
+		// TODO review how callers use these options to pass stdin.
+		// Probably they are not necessary, as they're related to
+		// attaching to the container (that is often unnecessary).
+		OpenStdin: rco.Stdin != nil,
+		StdinOnce: rco.Stdin != nil,
+		// TODO surprinsingly, we do not set AttachStdin...
+
+		AttachStdout: rco.Stdout != nil,
+	}
+}
+
+// asDockerHostConfig converts a RunContainerOptions into a HostConfig
+// understood by the docker client library.
+func (rco RunContainerOptions) asDockerHostConfig() docker.HostConfig {
+	hostConfig := docker.HostConfig{
+		CapDrop:         rco.CapDrop,
+		PublishAllPorts: rco.TargetImage,
+		NetworkMode:     rco.NetworkMode,
+	}
+	if rco.CGroupLimits != nil {
+		hostConfig.Memory = rco.CGroupLimits.MemoryLimitBytes
+		hostConfig.MemorySwap = rco.CGroupLimits.MemorySwap
+		hostConfig.CPUShares = rco.CGroupLimits.CPUShares
+		hostConfig.CPUQuota = rco.CGroupLimits.CPUQuota
+		hostConfig.CPUPeriod = rco.CGroupLimits.CPUPeriod
+	}
+	return hostConfig
+}
+
+// asDockerCreateContainerOptions converts a RunContainerOptions into a
+// CreateContainerOptions understood by the docker client library.
+func (rco RunContainerOptions) asDockerCreateContainerOptions() docker.CreateContainerOptions {
+	config := rco.asDockerConfig()
+	hostConfig := rco.asDockerHostConfig()
+	return docker.CreateContainerOptions{
+		Name:       "",
+		Config:     &config,
```

Comment:
@mfojtik we could do what you propose for `asDockerConfig` and `asDockerHostConfig`, but then for `asDockerCreateContainerOptions` and `asDockerAttachToContainerOptions` we don't want them to return pointers (for similar reasons in terms of "convenience of use").

That would lead to similar methods some returning pointers, some not.

WDYT of that situation?